### PR TITLE
Members feature

### PR DESCRIPTION
**Pull request by LoxiasCam**
# Members Only

For this project, we built a website where members can write posts. When logged in, members can see who the author of a post is. But when they are not logged in, they can only see the story and wonder who wrote it.

## User model created with devise:

Associations: One user can have many posts.
Authentication: Users can register and login into the website.

## Posts model
Posts have the following columns:
*    user_id
 *   Article

CRUD functionality: Users can see other posts even when they're not logged in. However, they may only create, edit and delete their own posts.
Associations: One post belongs to one user.
Secrets: Users may only see the author's email of a post when they are logged in.

### CODE REVIEW

**Comment by mke2111** [comment]
## Project Under review ⚪

**Review by mke2111 — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED 🔴 ♻️ 

## Hey Happy Coders, 👋🏽 
**This is a great project you have going on, well-done lads**. 

### Well-done 💯 
    1. The linter checks are successful
    2. The MVC is well structured.
    3. Thanks for the styles for the app.

**However, there are a few sections to improve before your project is approved.** 🔴 
<hr>

### REQUIRED

- [x] Add the migration command to your readme file in the getting started section. This is part of the sections that make a descriptive readme file.
- [x] Check the in-line comments for more sections to improve.

#

### Kindly make all the requested changes and submit another CRR. 💯 

**Cheers 🍹 and Happy coding 💻** 

<hr>

Mention @mke2111 in this thread and I'll be glad to help. 😄 
### Best regards
Slack 🆘  Roy Mukuye
Twitter 🐦  [Roy Mukuye](https://twitter.com/Roymkenya)
LinkedIn 🔗  [Roy Mukuye](https://www.linkedin.com/in/roy-mukuye/)

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    @post = Post.new
+  end
+
+  # GET /posts/1 or /posts/1.json
+  def show; end
+
+  # GET /posts/new
+  def new
+    @post = current_user.posts.build
+  end
```
Comment (mke2111) [inline app/controllers/posts_controller.rb:17]: - [x] Initialize a new empty post with `@post = Post.new`. [Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_new_requirement.png)

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -7,10 +7,75 @@
...
 
     <%= stylesheet_link_tag 'application', media: 'all', 'data-turbolinks-track': 'reload' %>
+    <%= stylesheet_link_tag "https://maxcdn.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css" %>
     <%= javascript_pack_tag 'application', 'data-turbolinks-track': 'reload' %>
   </head>
 
   <body>
+
+    <% if flash[:notice] %>
+      <div class="notification is-primary global-notification">
```
Comment (mke2111) [inline app/views/layouts/application.html.erb:17]: OPTIONAL

- [ ] It is a good practice to have zero logic in the views directory. Create methods in the helpers directory and call the methods in the views directory instead.

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -7,10 +7,75 @@
...
+
+      <div id="navbarExample" class="navbar-menu">
+        <div class="navbar-end">
+        <div class="navbar-item">
+          <div class="field is-grouped">
+            <p class="control"> 
+              <%= link_to "New Post", new_post_path, class: "button is-danger is-inverted" %>
+            </p>
+
+            <% if user_signed_in? %>
```
Comment (mke2111) [inline app/views/layouts/application.html.erb:50]: Another logic statement.

File: app/views/posts/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+        <% @posts.each do |post| %>
+          <div class="box">
+          <article class="media"> 
+            <div class="media-content"> 
+              <div class="content"> 
+                <%= post.article %>
+
+                <% if user_signed_in? && current_user.id == post.user_id %>
+
```
Comment (mke2111) [inline app/views/posts/index.html.erb:15]: Another logic statement

File: config/routes.rb [hunk 5]
```diff
@@ -1,3 +1,8 @@
+# frozen_string_literal: true
+
 Rails.application.routes.draw do
+  devise_for :users, controllers: { registrations: 'registrations' }
+  resources :posts
```
Comment (mke2111) [inline config/routes.rb]: - [x] Maintain these actions only `resources :posts, only: [:new, :create, :index]` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_routes_requirement.png))

**Review by jubaan — CHANGES_REQUESTED** [review]
[![Microverse][microverse-shield]][microverse-url]
<div>
<h1 align="center"> ⚠️ 🛠️ ⚠️ 🛠️ Changes Required 🛠️ ⚠️ 🛠️ ⚠️ </h1>
<br />
<p><b>"Fail fast, fail often"</b> and don't let this discourage you, because <b>only by failing do we learn to succeed</b>.</p>
<details open="open">
  <summary>Review Summary</summary>
  <ol>
    <li><a href="#congrats"> 💯 Awesome work 👏🏼 👏🏼 </a></li>
    <li><a href="#mandatory"> ⚠️ Things to Improve </a> 🚨 🚨 </li>
    <li><a href="#optional"> 💡Nice to have and other comments ✍🏼 ✍🏼 </a></li>
    <li><a href="#contact"> 🧔🏼 About your Reviewer </a> (you don't know it... but we are just automated bots 🤖) </li>
    <li><a href="#help"> 🆘 <b> 😭 Feeling stuck? or Need help</b>? 🙏🏼 🙏🏼 </a></li>
  </ol>
</details>
</div>

<h3 id="congrats"> 1. :100: Awesome work</h3>
<details close="close">
<summary> [click to open] Big or little achievements, but achievements at the end. <b>Feel proud that you're learning and improving everyday.</b></summary>



</details>

<hr>
<h3 id="mandatory"> 2. ⚠️ Things to Improve (or the To-Do list to get the project approved) </h3> 
<details open="open">
<summary> [click to open] Working on these observations will help you advance and get closer to the goal 🎯, postpone them or do them half-ways and they'll haunt you 👻 👻... FOREVER!! (It will feel like that, trust me). </summary>


- [ ] Review `Post#new` method in controller, and remove the unnecessary `@post`


</details>
<p><b>NOTE</b>This list is or could be divided in what is show in this comment and inline comments posted below.<p>
<hr>

<h3 id="optional"> 3. 💡 Nice to have and other comments </h3>
<details open="open">
<summary> [click to open] </h3></summary>


- [ ] Remove logic from views


</details>

<h3 id="contact"> 4. 🧔🏼 About your Reviewer </h3>
<details close="close">
<summary> [click to open] Say "hello world!" on Slack: @jubaan</summary>
<br />
<div align="center">
  <a src="https://www.github.com/jubaan">
    Julio Añoveros
  </a>
  <p>
    Julio Añoveros | @jubaan</br></br>
    <b>Full-stack Web Developer</b> and</br>
    🤖 Code Reviewer at <a href="https://www.microverse.org/?grsf=kfm2ej">Microverse</a>:rocket:
  </p>
</div>
</details>

<h3 id="help"> 5. 🆘 <b> 😭 Feeling stuck? or Need help</b>? 🙏🏼 🙏🏼</h3>
<details close="close">
  <summary> [click to open] <b>Feel stuck?</b> </summary>
  <br>
  <p> First breath... just do it. Take 15-20 min away from the computer, stretch, meditate, do some exercise, drink water, or do whatever is necessary to distract your mind from the obstacle.</p>
  <p><b>Sometimes you just need to get away to see things with a new perspective.</b></p>
</details>

<details close="close">
  <summary> [click to open] <b>Need help?</b> </summary>
  <br>
  <p> Send me a DM in <a href="https://www.microverse.org/?grsf=kfm2ej">Microverse</a> Students Slack, you'll find me as @jubaan. If I'm not available there are other Code Reviewers you can contact. They'll help as well.</p>
  <p>Remember that <a href="https://www.microverse.org/?grsf=kfm2ej">Microverse</a> has other channels where you can get an answer to your questions:</p>
  <ol>
    <li>The <b>students channel 🧑‍🎓 </b> in slack,</li>
    <li>your <b>teammates 👥 </b>,</li>
    <li>your <b>mentor 👨🏼 👩🏼 </b>, and</li>
    <li><b>Student Success 🏫 </b>.</li>
  </ol>
  <p><b> DON'T hesitate to ask! </b></p>
</details>
<p>If all of the above doesn't help... get a <a href="https://rubberduckdebugging.com/">rubber duck</a> ... it would be great to own one 😄. </p>

<h2 align="center"> Keep up the great work :muscle: and Happy Coding :smiley: :computer:!! </h1>

[microverse-shield]: https://img.shields.io/badge/-microverse-%236F23FF?style=for-the-badge
[microverse-url]: https://www.microverse.org/?grsf=kfm2ej
[bootstrap-shield]: https://img.shields.io/badge/Bootstrap-white?style=for-the-badge&logo=bootstrap
[bootstrap-url]: https://www.microverse.org/?grsf=kfm2ej

**Review by BrianSammit — APPROVED** [review]
STATUS: APPROVED 🟢 
-----------
Hello team great job working on the required changes 👏 👍 
-----------
Now you can merge this branch 💯 🥇 
HAPPY CODING!!! 

tenor (1)

**Pull request merged**